File: src/main/java/de/ddkfm/sparkdemo/ExampleWithTemplates.java
package de.ddkfm.sparkdemo;

import spark.ModelAndView;
import spark.Spark;
import spark.template.velocity.VelocityTemplateEngine;

import java.util.HashMap;
import java.util.Map;

public class ExampleWithTemplates {
    public static void main(String[] args) {
        Spark.get("/calculator",(req, resp) -> {
            resp.redirect("/calculator/add");
            return "";
        });
        Spark.get("/calculator/add", (req, resp) -> {
            Map<String, Object> model = new HashMap<>();
            model.put("calculationPath", "add");
            model.put("calculationMethod", "Addieren");
            return render(model, "templates/index.vm");
        });
        Spark.post("/calculator/add", (req, resp) -> {
            try {
                double a = Double.parseDouble(req.queryParams("a"));
                double b = Double.parseDouble(req.queryParams("b"));
                double c = a + b;
                Map<String, Object> model = new HashMap<>();
                model.put("calculationPath", "add");
                model.put("calculationMethod", "Addieren");
                model.put("result", c);
                return render(model, "templates/index.vm");
            } catch (NumberFormatException e) {
                resp.redirect("/calculator");
            }
            return "";
        });
    }
    private static String render(Map<String, Object> model, String templatePath) {
        return new VelocityTemplateEngine().render(new ModelAndView(model, templatePath));
    }
}

File: src/main/kotlin/de/ddkfm/sparkdemo/ExampleWithTemplates.kt
package de.ddkfm.sparkdemo

import spark.ModelAndView
import spark.Spark
import spark.template.velocity.VelocityTemplateEngine

fun main(args : Array<String>) {
    Spark.get("/calculator") {_,resp -> resp.redirect("/calculator/add")}
    Spark.get("/calculator/add") {req, resp ->
        val model = mapOf(
            "calculationPath" to "add",
            "calculationMethod" to "Addieren"
        )
        render(model, "templates/index.vm")
    }
    Spark.post("/calculator/add") {req, resp ->
        val a = req.queryParams("a")?.toDoubleOrNull()
        val b = req.queryParams("b")?.toDoubleOrNull()
        if(a == null || b == null) {
            resp.redirect("/calculator")
            return@post ""
        }
        val c = a + b
        val model = mapOf(
            "calculationPath" to "add",
            "calculationMethod" to "Addieren",
            "result" to c
        )
        render(model, "templates/index.vm")

    }
}

fun render(model : Map<String, Any>, templatePath : String) : String {
    return VelocityTemplateEngine().render(ModelAndView(model, templatePath))
}